doc/nrf/ug_memfault.rst

```diff
@@ -0,0 +1,207 @@
+
+1. Add the following Kconfig option in your :file:`prj.conf` file:
+
+   .. code-block:: console
+
+      CONFIG_MEMFAULT=y
+      CONFIG_MEMFAULT_NCS_PROJECT_KEY=""
+
+   The APIs in the Memfault SDK can then be linked into your application.
+   You can get your project key after signing up with Memfault.
+   The :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option is located in the :file:`../overlay-memfault.conf` file.
+
+#. Create a new folder in your project called :file:`config` and add the following three files:
+
+   * :file:`memfault_platform_config.h` - Sets Memfault SDK configurations that are not covered by Kconfig options.
+   * :file:`memfault_metrics_heartbeat_config.def` - Defines application-specific metrics.
+   * :file:`memfault_trace_reason_user_config.def` - Defines application-specific trace reasons.
+
+   Memfault SDK requires the three files in the include path during the build process.
+
+#. Add the following code in the :file:`CMakeLists.txt` file to have these configuration files in the include path:
+
+   .. code-block:: console
+
+      zephyr_include_directories(config)
+
+See the `Memfault nRF Connect SDK integration guide`_ documentation for more details.
+
+.. note::
+   In the :ref:`memfault_sample` sample folder, the aforementioned Memfault configurations are already included.
+   Make sure to add your project key in the :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option.
+
+You can also add configuration using multiple overlay files, ``-DOVERLAY_CONFIG`` must be set to a list of overlay configurations.
+See the following example on how to set an overlay configuration for Memfault integration:
+
+.. code-block:: console
+
+   west build -b nrf9160dk_nrf9160_ns -- -DOVERLAY_CONFIG=overlay-memfault.conf
+
+.. _ug_memfault_error:
+
+Error Tracking
+==============
+
+Memfault captures all faults that happen at any time.
+The crash data is sent to the Memfault cloud for further inspection and analysis.
+The faults are tracked using the following methods.
+
+Reboot reason tracking
+----------------------
+
+The device might reboot due to a crash, a brownout, or a firmware update.
+The reboot events for each device are captured and displayed in the Memfault UI.
+
+See `Memfault: Reboot tracking`_ for detailed documentation.
+
+Error tracking with trace events
+--------------------------------
+
+You can use the trace event method if you want to track errors without capturing full coredump traces (that use less storage) and keep the device up and running.
+The following samples implement a user-defined trace reason for demonstration purposes.
+
+* :ref:`memfault_sample`
+* :ref:`peripheral_mds`
+
+Read the **Error Tracking with trace events** section of these sample documents for more information.
+
+See `Memfault: Error Tracking with Trace Events`_ for detailed documentation.
+Faults are captured and displayed on the **Issues** page of Memfault UI.
```

Comment:
It's useful that we are this specific in where to find stuff on Memfault's UI, we just need to be sure to keep it up to date if they changw things